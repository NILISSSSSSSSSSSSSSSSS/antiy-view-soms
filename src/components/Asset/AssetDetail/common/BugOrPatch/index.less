.asset-detail-val{
  // background-color: #eef1f6;
  .back-btn{
    cursor: pointer;
    .back-result-icon{
      margin-left:5px;
    }
  }
  .radio-group-filter{
    margin-bottom:20px; 
  }
}
