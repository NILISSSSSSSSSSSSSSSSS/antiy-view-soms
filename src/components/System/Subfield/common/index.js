export default {
  deWeight: (item)=>{ //去除重复 级别的模块
    let obj = {}
    item = item.reduce((cur, next) => {
      if(!obj[next.levelType])
        obj[next.levelType] = true && cur.push(next)
      return cur
    }, [])
    return item
  },
  findPathByLeafId (leafId, nodes, next) { //遍历查找ID的父级
    let _t = this
    if(next === undefined) {
      next = []
    }
    for(let i = 0; i < nodes.length; i++) {
      let now = next.concat()  //浅拷贝
      now.push( nodes[i])
      if(leafId === nodes[i].stringId) {
        return now
      }
      if(nodes[i].childrenNode) {
        let findResult = _t.findPathByLeafId(leafId, nodes[i].childrenNode, now)
        if(findResult) {
          return findResult
        }
      }
    }
  }
}