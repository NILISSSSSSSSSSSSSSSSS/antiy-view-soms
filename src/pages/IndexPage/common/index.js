import { Fragment } from 'react'

const ASSETS_STATUS = [{
  name: '资产登记',
  value: 1
}, {
  name: '模板实施',
  value: 3
}, {
  name: '结果验证',
  value: 4
}, {
  name: '入网实施',
  value: 5
}, {
  name: '安全检查',
  value: 7
}, {
  name: '安全整改',
  value: 8
}, {
  name: '退役发起',
  value: 6
}, {
  name: '退役实施',
  value: 10
}]

export function onRow (ob, record) {
  let assetId, configId, bugpatchId, bugpatchInfoId = ''
  if (!record.businessId.includes('&') && record.businessId) {
    assetId = record.businessId //资产跳转
  } else if (record.businessId.split('&').length >= 3) {
    let len = record.businessId.split('&').length
    configId = record.businessId.split('&')[len - 1] //配置跳转
  } else if (['漏洞退回', '补丁处置', '补丁退回'].includes(record.name)) {
    bugpatchInfoId = record.businessId.split('&')[1]
  } else {
    bugpatchId = record.businessId.split('&')[0]//漏洞补丁跳转
  }
  switch (record.name) {
    case '基准配置':
      ob.push('/basesetting/manage?businessId=' + configId)
      break
    case '基准加固':
      ob.push('/basesetting/list/validation?businessId=' + configId)
      break
    case '基准核查':
      ob.push('/basesetting/list/enforcement?businessId=' + configId)
      break
    case (ASSETS_STATUS.some(item => item.name === record.name) ? record.name : ''):
      let init = ASSETS_STATUS.filter(item => item.name === record.name)[0].value
      ob.push(`/asset/manage?assetStatusList=${init}&conditionShow=true&id=` + assetId)
      break
    case (record.name.indexOf('装机模板') > -1 ? record.name : ''):
      ob.push('/asset/installtemplate?id=' + assetId)
      break
    case (['漏洞修复', '漏洞处置'].includes(record.name) ? record.name : ''):
      ob.push('/bugpatch/bugmanage/dispose?id=' + bugpatchId)
      break
    case '漏洞退回':
      ob.push('/bugpatch/bugmanage/information?id=' + bugpatchInfoId)
      break
    case '补丁安装':
      ob.push('/bugpatch/patchmanage/install?stringid=' + bugpatchId)
      break
    case '补丁处置':
      ob.push('/bugpatch/patchmanage/information?stringid=' + bugpatchInfoId)
      break
    case '补丁退回':
      ob.push('/bugpatch/patchmanage/information?stringid=' + bugpatchInfoId)
      break
    default:
      break
  }
}
//筛选流程类型
export function filterType (iData) {
  iData.forEach(item => {
    let init = item.processDefinitionId.split(':')[0]
    if (['assetAdmittance', 'installTemplate', 'assetRetire'].includes(init)) {
      item.typeName = '资产管理'
    } else if (['baselineConfig'].includes(init)) {
      item.typeName = '配置管理'
    } else if (['vulRepair'].includes(init)) {
      item.typeName = '漏洞管理'
    } else if (['patchInstall'].includes(init)) {
      item.typeName = '补丁管理'
    }
  })
  return iData
}
//大于四个字符返回万为单位
export const CharRegroup = (v)=>{
  if(!v) return 0
  v = String(v)
  if(v.length > 4) v = (
    <Fragment>
      {v.slice(0, v.length - 4)}
      <span style={{ display: 'inline-block', fontSize: '16px' }}>万</span>
    </Fragment>
  )
  return v
}

//图表配置
export const doughnutChart = {
  Dchart1: {
    order: 1,
    id: 'circle-chart-1',
    color: ['#3B6CFF', '#5BCDFF', '#AD7AFF', '#05CC7F', '#FFC438'],
    service: 'getHardware',
    title: '资产类型分布'
  },
  Dchart2: {
    order: 2,
    id: 'circle-chart-2',
    color: ['#4083FF', '#46ECFF', '#A569FF'],
    service: 'getAssetLevel',
    title: '资产重要程度分布'
  },
  Dchart3: {
    order: 3,
    id: 'circle-chart-3',
    color: ['#C22A4E', '#FF8E64', '#FFDA38', '#4083FF'],
    service: 'getWarningData',
    title: '告警等级分布'
  },
  Dchart4: {
    order: 4,
    id: 'circle-chart-4',
    color: ['#4083FF', '#46ECFF', '#A569FF', '#22B37A', '#FFDA38', '#FF8E64', '#4545FF'],
    service: 'getWarningDataChart',
    title: '告警类型分布'
  }
}