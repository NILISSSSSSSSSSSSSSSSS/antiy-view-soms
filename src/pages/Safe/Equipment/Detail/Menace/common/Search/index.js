import { Component } from 'react'
import CommonSearch from '@/components/common/Search'

const type = {
  apt: 'apt', // 高级威胁
  virus: 'virus', // 病毒威胁
  tanhai: 'tanhai'  //探海威胁
}
export default class Search extends Component {
  constructor (props) {
    super(props)
    this.state = {}
    this.defaultFields = [
      { label: '检索条件', key: 'retrieve', type: 'input', placeholder: 'IP/端口', data: [], maxLength: 200 },
      { label: '时间范围', key: 'time', type: 'dateRange', placeholder: ['开始日期', '结束日期'], data: [ ] }
    ]
  }
  // 重置查询
  handleReset = () => {
    const { reset, search } = this.props
    if(reset){
      reset()
    }else {
      search({})
    }
  }
  // 提交查询
  handleSubmit = (values = {}) => {
    let params = { ...values }
    if(values.time && values.time.length > 0) {
      const [ beginTime, endTime ] = values.time
      params.beginTime = beginTime ? beginTime.valueOf() : ''
      params.endTime = endTime ? endTime.valueOf() : ''
      delete params.time
    }
    params.multipleQuery = values.retrieve
    const { search } = this.props
    search && search(params)
  }
  /**
   * 给查询条件设置查询选项值
   * @param fields
   * @param list
   * @return {any[]}
   */
  setFilterData = (fields = [], list = []) => {
    return fields.map((e)=>{
      const currentData = list.find(it=>it.key === e.key) || {}
      if(e.key === currentData.key){
        return { ...e, ...currentData, data: currentData.data }
      }
      return e
    })
  }
  render () {
    const { placeholder, from } = this.props
    const { retrieve } = placeholder || {}
    let label = ''
    // 高级威胁
    if (from === type.apt){
      label = '追溯时间'
    }else if (from === type.virus){
      label = '中毒时间'
    }else if (from === type.tanhai){
      label = '最后活跃时间'
    }
    const fields = this.setFilterData(this.defaultFields, [{ key: 'retrieve', placeholder: retrieve }, { key: 'time', label }])
    return (
      <div className="search-bar">
        <CommonSearch wrappedComponentRef={search => { search && (this.searchForm = search.props.form) }} defaultFields={fields} onSubmit={this.handleSubmit} onReset={this.handleReset}/>
      </div>
    )
  }
}
