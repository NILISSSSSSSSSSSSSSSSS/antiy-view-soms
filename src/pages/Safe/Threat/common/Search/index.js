import { Component } from 'react'
import api from '@/services/api'
import CommonSearch from '@/components/common/Search'

const type = {
  apt: 'apt', // 高级威胁
  virus: 'virus', // 病毒威胁
  tanhai: 'tanhai'  //探海威胁
}

export default class Search extends Component {
  constructor (props) {
    super(props)
    this.state = {
      equList: [] // 所属的安全设备列表
    }
    this.filterFields = [
    ]
    this.defaultFields = [
      { label: '所属安全设备', allowClear: false, placeholder: '全部', key: 'safetyId', type: 'select', config: { name: 'name', value: 'id' } },
      { label: '检索条件', key: 'retrieve', type: 'input', placeholder: '全部', data: [], maxLength: 200 },
      { label: '时间范围', key: 'time', type: 'dateRange', placeholder: ['开始日期', '结束日期'], data: [ ] }
    ]
  }
  componentDidMount () {
    const { from } = this.props
    // 高级威胁
    if (from === type.apt){
      this.getAPTEqu()
    }else if (from === type.virus){
      this.getVirusEqu()
    }else if (from === type.tanhai){
      this.getTanhaiEqu()
    }
  }

  // 获取高级威胁所属的安全设备
  getAPTEqu = () => {
    api.getsafetythreataptEquip().then((res)=>{
      if(res){
        this.setState({ equList: res.body })
      }
    })
  }
  // 获取探海威胁所属的安全设备
  getTanhaiEqu = () => {
    api.getsafetythreattanhaiEquip().then((res)=>{
      if(res){
        this.setState({ equList: res.body })
      }
    })
  }

  // 获取病毒所属的安全设备
  getVirusEqu = () => {
    api.getsafetythreatvirusEquip().then((res)=>{
      if(res){
        this.setState({ equList: res.body })
      }
    })
  }
  // 重置查询
  handleReset = () => {
    const { reset } = this.props
    reset && reset()
  }

  handleSubmit = (values = {}) => {
    let params = { ...values }
    if(values.time && values.time.length > 0) {
      const [ beginTime, endTime ] = values.time
      params.beginTime = beginTime ? beginTime.valueOf() : ''
      params.endTime = endTime ? endTime.valueOf() : ''
      delete params.time
    }
    params.safetyId = values.safetyId
    params.multipleQuery = values.retrieve
    const { search } = this.props
    search && search(params)
  }
  /**
   * 给查询条件设置查询选项值
   * @param fields
   * @param list
   * @return {any[]}
   */
  setFilterData = (fields = [], list = []) => {
    return fields.map((e)=>{
      const currentData = list.find(it=>it.key === e.key) || {}
      if(e.key === currentData.key){
        return { ...e, ...currentData, data: currentData.data }
      }
      return e
    })
  }
  render () {
    const { placeholder, from } = this.props
    const { safetyId, retrieve } = placeholder || {}
    const { equList } = this.state
    let label = ''
    // 高级威胁
    if (from === type.apt){
      label = '追溯时间'
    }else if (from === type.virus){
      label = '中毒时间'
    }else if (from === type.tanhai){
      label = '最后活跃时间'
    }
    const fields = this.setFilterData(this.defaultFields, [{ key: 'safetyId', data: equList, placeholder: safetyId }, { key: 'time', label }, { key: 'retrieve', placeholder: retrieve }])
    return (
      <div className="search-bar">
        <CommonSearch wrappedComponentRef={search => { search && (this.searchForm = search.props.form) }} defaultFields={fields} onSubmit={this.handleSubmit} onReset={this.handleReset}/>
      </div>
    )
  }
}
