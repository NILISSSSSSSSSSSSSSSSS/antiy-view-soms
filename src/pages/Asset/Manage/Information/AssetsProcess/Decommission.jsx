import React, { Component, Fragment } from 'react'
import { connect } from 'dva'
import { CommonModal } from '@c'
import { Form, Radio, Input, Upload, Icon, message, Table } from 'antd'
import PropTypes from 'prop-types'
import { uploadLimit, createHeaders, TooltipFn } from '@/utils/common'
import { cloneDeep, throttle } from 'lodash'
import './style.less'
import api from '@/services/api'

import ProBasic from'./ProcessCom/ProBasic'

const FormItem = Form.Item
const { TextArea } = Input
const RadioGroup = Radio.Group

@Form.create()
@connect(({ asset, staticAsset }) => ({
  assetModelColumns: staticAsset.assetModelColumns,
  previousBatch: asset.previousBatch
}))
class Decommission extends Component{
    state = {
      PrefixCls: 'AssetsProcess',
      isReject: false, //是否必填
      fileList: '',
      visible: false,
      isTable: true,
      recordsArr: [],
      keyWords: '退役',
      assetStatus: 9,
      basicObj: {
        remarkName: '',
        note: '',
        fileInfo: '[]'
      }  //上一步附件等信息
    }
    uuids=[]
    cloneInitState={}  //保存初始状态

    static defaultProps= {

    }

    static propTypes = {
      assetModelColumns: PropTypes.array,
      previousBatch: PropTypes.array,
      onRef: PropTypes.func
      // assetStatus: PropTypes.oneOf([9, 13])
    }

    openModel=(visible = true, recordsArr = [], assetStatus = 9)=>{
      const Arr = [
        { name: '退役', val: 9 },
        { name: '报废', val: 13 }
      ]
      this.setState({
        visible,
        recordsArr,
        assetStatus,
        keyWords: Arr.filter(item=>item.val === assetStatus)[0].name,
        isTable: recordsArr.length > 1 ? true : false
      }, ()=>{
        if(!visible) return void(0)
        Promise.all([
          this.previousBatch()
        ])
      })
    }

    //上传附件
    UploadChange = (info) => {
      const fileList = info.fileList.slice(0, 5)
      this.setState({ fileList })
      // 网络失败
      fileList.filter((e)=>e.status === 'error').forEach((file)=>{
        message.info(file.name + '上传超时！')
      })
      //服务器处理失败
      fileList.filter((e)=>e.status === 'done' && e.response && e.response.head.code !== '200' && e.percent === 100 && !this.uuids.includes(e.uid)).forEach((file)=>{
      // 如果提示上传失败，下次上传时，不在提示
        this.uuids.push(file.uid)
        message.info('上传 ' + file.name + ' 失败！')
      })
      // 所有文件上传完成时，排除上传失败文件，刷新上传列表
      const list = fileList.filter((e)=>e.status === 'done')
      if(list.length === fileList.length){
        this.uuids = []
        this.setState({
          fileList: list.filter((e)=>e.response && e.response.head.code === '200')
        })
      }
    }

    //ui同意拒绝回调
    changeReject = () => {
      const isAllow = this.props.form.getFieldValue('agree') === '0' || this.props.form.getFieldValue('agree') === 0
      this.setState({
        isReject: !isAllow
      })
    }

    getRecordsArrIds=(stringId)=>{
      let stringIds = []
      this.state.recordsArr.forEach(v=>stringIds.push(v[stringId]))
      return stringIds
    }

    /***************副作用开始************** */

    previousBatch=async ()=>{
      await  this.props.dispatch({ type: 'asset/previousBatch', payload: { ids: this.getRecordsArrIds('stringId') } })
      const { previousBatch } = this.props
      const isAdd = !!previousBatch.length
      const note = isAdd ? previousBatch[0].note : ''
      const fileInfo = isAdd ? previousBatch[0].fileInfo : '[]'
      this.setState({
        basicObj: {
          remarkName: `${this.state.keyWords}方案`,
          note,
          fileInfo
        }
      })
    }

    //提交
    onSubmit = throttle(values => {
      this.props.form.validateFields((err, values = {}) => {
        if (err) return void(0)

        const fileList = values.file && values.file.fileList
        let fileInfo = []
        if(fileList) {
          fileList.forEach((item)=>{
            const obj = {
              url: item.response.body[0].fileUrl,
              fileName: item.response.body[0].originFileName
            }
            fileInfo.push(obj)
          })
        }
        fileInfo = JSON.stringify(fileInfo)
        let assetInfoList = []
        this.state.recordsArr.forEach(item=>{
          const content = {
            taskId: item.waitingTaskReponse.taskId,
            assetId: item.stringId
          }
          assetInfoList.push(content)
        })

        const params = {
          agree: values.agree,
          assetFlowEnum: 'RETIRE',
          assetInfoList,
          formData: { 'assetRetireResult': values.agree },
          fileInfo: fileInfo,
          note: values.note

        }

        api.statusJump(params).then(res=>{
          if (res && res.head && res.head.code === '200') {
            this.props.getAssetList()
            this.openModel(false)
            message.success(`${res.head.result}`)
          }
        })
      })
    }, 5000, { trailing: false })

    render (){
      const { PrefixCls, isReject, fileList, visible, basicObj, isTable, recordsArr, keyWords } = this.state
      const {
        assetModelColumns,
        form: { getFieldDecorator }
      } = this.props

      let columns = cloneDeep(assetModelColumns)

      const operateColumns = [
        {
          title: 'MAC',
          dataIndex: 'note',
          key: 'note',
          render: text=>TooltipFn(text)
        },
        {
          title: `${keyWords}方案`,
          dataIndex: 'note1',
          key: 'note1',
          render: text=>TooltipFn(text)
        },
        {
          title: '附件',
          dataIndex: 'fileInfo',
          key: 'fileInfo',
          render: text=>{
            const fileInfos = text ? JSON.parse(text) : []
            const styles = {
              style: {
                display: 'block',
                overflow: 'hidden',
                'text-overflow': 'ellipsis',
                'white-space': 'nowrap'
              }
            }
            return Array.isArray(fileInfos) && fileInfos.map((v, i)=>(
              <a {...styles} onClick={() => window.open(`/api/v1/file/download?access_token=${sessionStorage.getItem('token')}&url=${v.url}&fileName=${encodeURIComponent(v.fileName)}`)}>
                {/* {TooltipFn(v.fileName)} */}
                {v.fileName}
              </a>
            ))
          }
        },   {
          title: '审批意见',
          dataIndex: 'note2',
          key: 'note2',
          render: text=>TooltipFn(text)
        }, {
          title: '操作',
          isShow: true,
          key: 'stringId',
          width: '20%',
          render: (record, item)=>{
            return(
              <Fragment>
                <span className="operate-wrap">
                  <a onClick={()=>this.lookReportBatch(item.stringId)}>查看</a>
                </span>
                <div className="operate-wrap">
                  <a onClick={()=>this.delete(item.stringId)}>移除</a>
                </div>
              </Fragment>
            )
          }
        }
      ]
      columns = columns.concat(operateColumns)

      const formItemLayout = {
        labelCol: {
          span: 5
        },
        wrapperCol: {
          span: 18
        }
      }

      const config = {
        name: 'fileList',
        action: '/api/v1/asset/file/upload',
        beforeUpload: file=>uploadLimit(file, fileList, 10),
        accept: '.rar,.zip,.7z,.doc,.docx,.jpg,.pdf,.png,.txt,.xlsx,.xls',
        onChange: this.UploadChange,
        fileList,
        data: {
          fileUse: 'SCHEME_FILE'
        },
        headers: createHeaders()
      }

      const hintContent = (
        <div>
          <p>支持上传大小：0-10M</p>
          <p>支持扩展名：.rar .zip .7z .doc .docx .jpg .pdf .png .txt .xlsx .xls </p>
        </div>
      )

      const ProBasicFn = cont=>(!!basicObj.note === false && (!!(JSON.parse(basicObj.fileInfo).length) === false))
        ? null : cont

      return(
        <CommonModal
          title={`${keyWords}执行`}
          type="search"
          visible={visible}
          width={isTable ? 900 : 680}
          oktext='确定'
          onConfirm={this.onSubmit}
          onClose={()=>this.openModel(false)}>
          <div className={PrefixCls}>
            <div className='process-wrap'>

              <div className="modal-content">
                {isTable ? <Table
                  className="table-small"
                  rowKey='stringId'
                  columns={columns}
                  dataSource={recordsArr}
                  pagination={false}/> :
                  ProBasicFn(<ProBasic basicObj={basicObj}/>)}
                <section className="operate-info">
                  <div className="operate-content">
                    <div className="result-info">
                      <img src={require('@/assets/item-icon.svg')} className="item-icon" alt=""/>
                      <span>审批意见</span>
                      <span className="info">同意!</span>
                    </div>
                  </div>
                </section>

                <section>
                  <h3 style={{ textAlign: 'center' }}>{keyWords}已完成</h3>
                  <Form layout="horizontal">
                    {/* <FormItem {...formItemLayout} label="审批意见">
                      {getFieldDecorator('agree', {
                        rules: [{ required: true, message: '请选择！' }],
                        initialValue: 1
                      })(
                        <RadioGroup onChange={this.changeReject}>
                          <Radio value={1}>通过</Radio>
                          <Radio value={0}>未通过</Radio>
                        </RadioGroup>
                      )}
                    </FormItem> */}
                    <FormItem {...formItemLayout} label="备注">
                      {getFieldDecorator('note', {
                        rules: [
                          { message: '最多300个字符！', max: 300 },
                          { required: isReject, message: '请输入备注！' }
                        ]
                      })(
                        <TextArea rows={6} placeholder="请输入..." style={{ resize: 'none' }} />
                      )}
                    </FormItem>
                    <FormItem  {...formItemLayout} label="上传附件">
                      {getFieldDecorator('file')(
                        <Upload {...config}>
                          <div className="work-order-add-upload">
                            <Icon type="plus" />
                             &nbsp;&nbsp;上传附件&nbsp;&nbsp;
                          </div>
                        </Upload>
                      )}
                      <div className="format">
                        {hintContent}
                      </div>
                    </FormItem>
                  </Form>
                </section>

              </div>
            </div>
          </div>

        </CommonModal>
      )
    }

    UNSAFE_componentWillMount (){
      this.cloneInitState = cloneDeep(this.state)
      delete this.cloneInitState.visible
    }

    componentDidMount (){
      this.props.onRef(this, 9)
    }

    shouldComponentUpdate (nextProps,  nextState){
      if((nextState.visible !== this.state.visible) && nextState.visible === false){
        // 重置state
        this.setState({ ...this.cloneInitState })
      }
      return true
    }
}

export default Decommission
