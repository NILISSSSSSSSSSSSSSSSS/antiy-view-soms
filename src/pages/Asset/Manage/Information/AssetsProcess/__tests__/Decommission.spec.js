import React from 'react'
import Decommission  from '../Decommission'
import { shallow } from 'enzyme'
import PropTypes from 'prop-types'
import { TooltipFn } from '@u/common'

const statusJump = (params)=> {
  return Promise.resolve({
    head: { code: '200' },
    body: { }
  })
}
const api = {
  statusJump
}

jest.mock('../../../../../../services/api')
const props = {
  onRef: jest.fn()
}

const store = {
  subscribe: () => ({
  }),
  dispatch: () =>({}),
  getState: () => ({
    asset: {
      previousBatch: [
        {
          note: '',
          fileInfo: ''
        }
      ]
    },
    staticAsset: {
      assetModelColumns: [
        {
          title: '资产名称',
          dataIndex: 'name',
          key: 'name',
          render: text=>TooltipFn(text)
        },
        {
          title: 'IP',
          dataIndex: 'ip',
          key: 'ip',
          render: text=>TooltipFn(text)
        }
      ]
    }
  })
}
const options = {
  context: { store },
  childContextTypes: { store: PropTypes.object.isRequired }
}

describe('<Decommission />', () => {
  let wrapper, instance

  beforeEach(() => {
    wrapper = shallow(<Decommission {...props}/>).dive(options).dive(options)
    instance = wrapper.instance()
  })

  const testInit = ()=>{
    const recordsArr = [{
      stringId: 666,
      waitingTaskReponse: {
        taskId: 666
      }
    }]
    instance.openModel(true, recordsArr)
  }

  it('render <Decommission /> component', () => {
    expect(wrapper.find('.process-wrap')).toHaveLength(1)
  })

  it('test  shouldComponentUpdate', () => {
    const nextState1 = {
      visible: true
    }
    const nextState2 = {
      visible: false
    }
    instance.shouldComponentUpdate(null, nextState1)
    instance.state.visible = true
    instance.shouldComponentUpdate(null, nextState2)
  })

  it('testFunc openModel',  ()=> {
    instance.openModel(false)
  })

  it('testFunc UploadChange',  ()=> {
    const info = {
      fileList: [
        {
          name: 666,
          uid: 666,
          status: 'done',
          percent: 100,
          response: {
            head: {
              code: '200'
            }
          }
        }
      ]
    }
    instance.UploadChange(info)
  })

  it('testFunc changeReject',  ()=> {
    const spyFunc = jest.spyOn(instance, 'changeReject')
    spyFunc()
    expect(spyFunc).toHaveBeenCalled()
  })

  it('testFunc getRecordsArrIds',  ()=> {
    testInit()
    const spyFunc = jest.spyOn(instance, 'getRecordsArrIds')
    spyFunc('stringId')
    expect(spyFunc).toHaveBeenCalled()
  })

  it('testFunc previousBatch',  ()=> {
    const spyFunc = jest.spyOn(instance, 'previousBatch')
    spyFunc()
    expect(spyFunc).toHaveBeenCalled()
  })

  it('testFunc onSubmit',  ()=> {
    testInit()
    const spyFunc = jest.spyOn(instance, 'onSubmit')
    spyFunc()
    expect(spyFunc).toHaveBeenCalled()
  })

  it('shoulder render a Decommission',  ()=> {
    const app = wrapper.debug()
    expect(app).toMatchSnapshot()
  })

})