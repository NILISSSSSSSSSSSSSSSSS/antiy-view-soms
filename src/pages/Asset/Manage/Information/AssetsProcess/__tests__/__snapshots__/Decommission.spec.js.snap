// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Decommission /> shoulder render a Decommission 1`] = `
"<Connect(Form(commonModal)) title=\\"退役\\" type=\\"search\\" visible={false} width={680} oktext=\\"确定\\" onConfirm={[Function]} onClose={[Function: onClose]}>
  <div className=\\"AssetsProcess\\">
    <div className=\\"process-wrap\\">
      <div className=\\"modal-content\\">
        <section>
          <h3>
            填写结果
          </h3>
          <Form layout=\\"horizontal\\" colon={true} hideRequiredMark={false} onSubmit={[Function: onSubmit]}>
            <FormItem labelCol={{...}} wrapperCol={{...}} label=\\"退役情况\\" hasFeedback={false}>
              <RadioGroup onChange={[Function: bound bound onCollectValidate]} disabled={false} buttonStyle=\\"outline\\" value={1} id=\\"agree\\" data-__meta={{...}} data-__field={{...}}>
                <Radio value={1} type=\\"radio\\">
                  通过
                </Radio>
                <Radio value={0} type=\\"radio\\">
                  拒绝
                </Radio>
              </RadioGroup>
            </FormItem>
            <FormItem labelCol={{...}} wrapperCol={{...}} label=\\"备注\\" hasFeedback={false}>
              <TextArea rows={6} maxLength={300} placeholder=\\"填写退役实施结果，或拒绝退役的原因\\" style={{...}} value={[undefined]} id=\\"note\\" onChange={[Function: bound bound onCollectValidate]} data-__meta={{...}} data-__field={{...}} />
            </FormItem>
            <FormItem labelCol={{...}} wrapperCol={{...}} label=\\"上传附件\\" hasFeedback={false}>
              <Upload name=\\"fileList\\" action=\\"/api/v1/asset/file/upload\\" beforeUpload={[Function: beforeUpload]} accept=\\".rar,.zip,.7z,.doc,.docx,.jpg,.pdf,.png,.txt,.xlsx,.xls\\" onChange={[Function: bound bound onCollect]} fileList=\\"\\" data={{...}} headers={{...}} type=\\"select\\" multiple={false} showUploadList={true} listType=\\"text\\" className=\\"\\" disabled={false} supportServerRender={true} value={[undefined]} id=\\"file\\" data-__meta={{...}} data-__field={{...}}>
                <div className=\\"work-order-add-upload\\">
                  <Icon type=\\"plus\\" />
                    上传附件  
                </div>
              </Upload>
              <div className=\\"format\\">
                <div>
                  <p>
                    支持上传大小：0-10M
                  </p>
                  <p>
                    支持扩展名：.rar .zip .7z .doc .docx .jpg .pdf .png .txt .xlsx .xls 
                  </p>
                </div>
              </div>
            </FormItem>
          </Form>
        </section>
      </div>
    </div>
  </div>
</Connect(Form(commonModal))>"
`;
